Quick search menu in blockly editor

joueur883 opened the pull request:
**- Added a quick search menu in blockly editor (open with middle click, close by clicking anywhere in the editor)**

## Why add a quick search while there is already a search field ?
**It can be faster to directly searching the block you need directly with middle click on the mouse and searching instantly the needed block to code faster.**

## What it looks like in the editor :

<img width="1507" height="681" alt="Screenshot1" src="https://github.com/user-attachments/assets/8434326c-5f4d-4268-aec2-f9efa4567d7d" />

<img width="1501" height="679" alt="Screenshot2" src="https://github.com/user-attachments/assets/aecea93c-7c68-41c9-b008-bfff0105fd6a" />

Goldorion commented on src/main/java/net/mcreator/blockly/data/ToolboxBlock.java:
```diff
@@ -242,4 +242,4 @@ String getGroupEstimate() {
 	}
 
-}
+}
```
Not needed

Goldorion commented on src/main/java/net/mcreator/ui/blockly/BlocklyPanel.java:
```diff
@@ -74,6 +88,7 @@ public class BlocklyPanel extends JFXPanel implements Closeable {
 	private final MCreator mcreator;
 	private final BlocklyEditorType type;
 
+
```
It can be removed

Goldorion commented on src/main/java/net/mcreator/ui/blockly/BlocklyPanel.java:
```diff
@@ -173,6 +188,184 @@ public BlocklyPanel(MCreator mcreator, @Nonnull BlocklyEditorType type) {
 				}
 			});
+
+
```
One line is enough

Goldorion commented on src/main/java/net/mcreator/ui/blockly/BlocklyPanel.java:
```diff
@@ -173,6 +188,184 @@ public BlocklyPanel(MCreator mcreator, @Nonnull BlocklyEditorType type) {
+
+				if (event.getButton() == MouseButton.MIDDLE) {
+
+					class quickSearchMenuPopup extends JFrame {
```
Externalize the class as it's a pretty big one (and you should rename it for `QuickSearchMenuPopup`, so it's the same format as everything else.

Goldorion commented on src/main/java/net/mcreator/ui/blockly/BlocklyEditorToolbar.java:
```diff
@@ -61,7 +62,8 @@ public class BlocklyEditorToolbar extends TransparentToolBar {
 
-	private final BlocklyPanel blocklyPanel;
+	public final BlocklyPanel blocklyPanel;
+	public static BlocklyPanel staticBlocklyPanel = null;
```
Honestly, I'm not sure if using another Blockly panel on top of the main one is a good one, because one is already a strong performance eater on smaller computers, so having to load another one may cause problems with users.

KlemenDEV commented on src/main/java/net/mcreator/ui/blockly/BlocklyPanel.java:
```diff
@@ -173,6 +188,184 @@ public BlocklyPanel(MCreator mcreator, @Nonnull BlocklyEditorType type) {
+
+						}
+
+						private void updateQuickSearch(JTextField quickSearchField, JScrollablePopupMenu scrollMenu) {
```
You should reuse system of existing search bar on the top, not re-implement it

joueur883 commented:
I think this feature can be useful because it offers a contextual search menu that appears directly next to the mouse pointer.
It’s designed for speed and flow, especially when building procedures or adding multiple blocks quickly.

Here's a short demo (sorry for the mouse glitches, I don't have OBS):

https://github.com/user-attachments/assets/173e8cdb-ad14-4332-b915-87cebbfc56e6


What do you think of this quick search menu?

joueur883 commented:
I replaced the JFrame with a JDialog, externalized the QuickSearchMenuPopup class, and added a "recent searches" feature to the quick search menu. This allows users to quickly reuse the last 9 searched blocks, which I believe adds practical value for repetitive workflows.

<img width="1075" height="585" alt="recents" src="https://github.com/user-attachments/assets/8b8dc71e-ab29-4f16-befd-48c2f254a7b8" />

Regarding your question about its usefulness compared to double Shift: I see this as a complementary feature, especially for users (like me) who prefer mouse-driven interactions. The contextual popup appears right next to the cursor, which helps maintain flow when building procedures with the mouse.

Happy to answer any questions or suggestions you might have!

joueur883 commented:
Thanks for the feedback and review. I understand your concerns regarding code duplication and the added value of this feature. I’ll close this PR for now and keep the idea aside in case it fits better in a future iteration or context. Appreciate the time spent reviewing!
